src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/clients/CcdCaseEventsConsumer.java

```diff
@@ -42,26 +43,38 @@ public void run() {
     @SuppressWarnings({"PMD.DataflowAnomalyAnalysis"})
     protected void consumeMessage(ServiceBusSessionReceiverClient sessionReceiver) {
         try (ServiceBusReceiverClient receiver = sessionReceiver.acceptNextSession()) {
-            receiver.receiveMessages(1).forEach(
-                message -> {
-                    try {
-                        String messageId = message.getMessageId();
-                        String sessionId = message.getSessionId();
-                        log.info("Received CCD Case Event message with id '{}' and case id '{}'", messageId, sessionId);
+            if (receiver != null) {
+                receiver.receiveMessages(1).forEach(
+                    message -> {
+                        try {
+                            String messageId = message.getMessageId();
+                            String sessionId = message.getSessionId();
+                            log.info(
+                                "Received CCD Case Event message with id '{}' and case id '{}'",
+                                messageId,
+                                sessionId
+                            );
 
-                        eventMessageReceiverService.handleCcdCaseEventAsbMessage(messageId, sessionId,
-                                new String(message.getBody().toBytes()));
-                        receiver.complete(message);
+                            eventMessageReceiverService
+                                .handleCcdCaseEventAsbMessage(messageId, sessionId,
+                                                              new String(message.getBody().toBytes()));
+                            receiver.complete(message);
 
-                        log.info("CCD Case Event message with id '{}' handled successfully", messageId);
-                    } catch (Exception ex) {
-                        log.error("Error processing CCD Case Event message with id '{}' - "
-                                + "abandon the processing and ASB will re-deliver it", message.getMessageId());
-                        receiver.abandon(message);
-                    }
-                });
+                            log.info("CCD Case Event message with id '{}' handled successfully", messageId);
+                        } catch (Exception ex) {
+                            log.error(
+                                "Error processing CCD Case Event message with id '{}' - "
```

Comment:
thoughts on adding sampling for this exception? as well? if it's all the same do we need to log all of them to app insights? https://learn.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-sampling-overrides#getting-started

Reply:
Yeah I think Rafal is going to follow up at looking into that.  This is the first PR since we saw lots of NPEs.